test/orderController.test.js

```diff
@@ -0,0 +1,215 @@
+    const orderData = {
+      orderItems: [
+        {
+          album: album._id,
+          quantity: 2,
+        },
+      ],
+      subtotal: 100,
+      tax: 0.1,
+      total: 110,
+      user: user._id,
+    };
+
+    const response = await request(app).post(`/api/v1/orders`).send(orderData);
+    expect(response.status).toBe(StatusCodes.UNAUTHORIZED);
+  });
+  //create an order -Error case - some data is missing
+
+  it('should return a 400 status if order items are missing', async () => {
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+    const orderData = {
+      subtotal: 100,
+      tax: 0.1,
+      total: 110,
+      user: user._id,
+    };
+
+    const response = await request(app)
+      .post('/api/v1/orders')
+      .set('Cookie', signedCookie)
+      .send(orderData);
+
+    expect(response.status).toBe(StatusCodes.BAD_REQUEST);
+  });
+
+  // Delete order - Success case
+//   it('should delete an order successfully', async () => {
+//      // Create an order
+
+//     //create an admin
+//     admin = await User.create({
+//       email: 'Holly@google.com',
+//       password: 'secret',
+//       name: 'Holly',
+//       username: 'Holly123',
+//       role: 'admin',
+//     });
+//     const adminCredentials = {
+//       email: 'Holly@google.com',
+//       password: 'secret',
+//     };
+//     const signedCookie = await loginAndReturnCookie(adminCredentials);
+//      //Akos: get   TypeError: Order.create is not a function //maybe the error is in export in Order.js? but I've checked that it's exported correctly
+//         // Mock ID
+// const mockOrderId = new mongoose.Types.ObjectId();
+
+//      const orderData = {
+//       _id: mockOrderId,  // Assigning the mock ID
+//       user: admin._id,
+//       orderItems: [{ album: album._id, quantity: 2 }],
+//       subtotal: 200,
+//       tax: 0.1,
+//       total: 210,
+//     };
+//     const order = await Order.create(orderData);
+
+//     const response = await request(app)
+//       .delete(`/api/v1/orders/${order._id}`) // Assuming testUser is the user you want to delete
+//       .set('Cookie', signedCookie);
+
+//     expect(response.status).toBe(StatusCodes.OK);
+//     expect(response.body).toEqual({ msg: 'Success! Order was deleted' });
+//   });
+
+  // Delete order - Error case - Order not found
+  it('should return a 404 status if the order to delete is not found', async () => {
+    await Order.deleteMany({}); //  TypeError: Order.deleteMany is not a function
+    const nonExistingOrderId = 'nonexistingorderid';
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    const response = await request(app)
+      .delete(`/api/v1/orders/${nonExistingOrderId}`)
+      .set('Cookie', signedCookie);
+
+    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+  });
+
+  // Delete order - Error case - Unauthorized user
+  it('should return a 403 status if user is not authorized to delete the order', async () => {
```

Comment:
Actually, this test is surfacing something a bit strange (which is completely the purpose of writing tests, so great job!):

*   According to `orderRoutes.js` we only allow admins to delete orders using this endpoint: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/routes/orderRoutes.js#L22](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/routes/orderRoutes.js#L22)
*   But in the controller function for this endpoint, we're calling `checkPermissions` which is usually used to ensure that only an admin or the matching user can do this action: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/controllers/orderController.js#L145](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/controllers/orderController.js#L145)

So in this case, the test is passing because the request is indeed returning a 403 Forbidden. But it's not because it's a user trying to delete anothe user's credentials. If we add some log lines, we can see that the code doesn't even get that far, because the request is rejected first by the `authorizePermissions` middleware and doesn't execute the controller code at all. 

We can likely just reword this test to be about non-admin users not being authorized to delete the order. And in that case, we don't need to create `anotherUser`; we can just log in as `userCredentials` who is already not an admin and thus should receive a 403 Forbidden.

Reply:
@akosasante I deletes this checkPermissions(req.user, order.user);  from orderController.js 
it seems to have no sence since weonly let admin to delete the order?
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/d81a7ff1-8059-4b1a-893f-481a9058d8bd)

Reply:
Yep that makes sense to me :+1: